client/hooks/use-report-export.ts

```diff
@@ -0,0 +1,202 @@
+ * 	exportFileAvailabilityEndpoint: '/wc/v3/payments/transactions/download',
+ * 	userEmail: 'test@example.com',
+ * } );
+ */
+export const useReportExport = () => {
+	const [ isDownloading, setIsDownloading ] = useState( false );
+	const { createNotice } = useDispatch( 'core/notices' );
+	const timeoutIdRef = useRef< NodeJS.Timeout | null >( null );
+	const retryCountRef = useRef( 0 );
+
+	useEffect( () => {
+		// Cancel the timeout if the component unmounts.
+		return () => {
+			if ( timeoutIdRef.current ) {
+				clearTimeout( timeoutIdRef.current );
+			}
+		};
+	}, [ timeoutIdRef ] );
+
+	interface PollForFileProps {
+		/**
+		 * The URL to check for the exported file. E.g. "/wc/v3/payments/transactions/download/{export_id}"
+		 */
+		checkFileURL: string;
+		/**
+		 * The email address the export will be sent to.
+		 */
+		userEmail: string;
+		/**
+		 * The interval in milliseconds to poll the server.
+		 */
+		interval?: number;
+	}
+
+	/**
+	 * Polls the server to check if the exported file is ready.
+	 */
+	async function pollForFile( {
+		checkFileURL,
+		userEmail,
+		interval = 1000,
+	}: PollForFileProps ) {
+		timeoutIdRef.current = setTimeout( async () => {
+			retryCountRef.current++;
+			let exportedFileURLResponse;
+			try {
+				exportedFileURLResponse = await apiFetch< ExportURLResponse >( {
+					path: checkFileURL,
+					method: 'GET',
+				} );
+			} catch ( error ) {
+				// If there's an error (like 500), continue with retry logic
+				exportedFileURLResponse = { status: 'error' };
+			}
+
+			if (
+				'success' === exportedFileURLResponse.status &&
+				exportedFileURLResponse.download_url &&
+				exportedFileURLResponse.download_url.includes(
+					'exports.wordpress.com'
+				)
+			) {
+				// The file is available, so we can download it.
+				// Create a link element to trigger the download.
+				const link = document.createElement( 'a' );
+				// Add force_download=true to the URL to force the download, which adds the appropriate `Content-Disposition: attachment` header when using production server.
+				link.href =
+					exportedFileURLResponse.download_url +
+					'?force_download=true';
+				link.click();
+
+				recordEvent( 'wcpay_transactions_download_csv_in_browser' );
+
+				createNotice(
+					'success',
+					sprintf(
+						__(
+							'Your CSV export is ready to download. It will also be emailed to %s',
+							'woocommerce-payments'
+						),
+						userEmail
+					)
+				);
+
+				setIsDownloading( false );
+
+				return;
+			}
+
+			if ( retryCountRef.current < maxRetries ) {
+				// If the file is not available, check again after 1 second.
+				pollForFile( {
+					checkFileURL,
+					userEmail,
+				} );
+			} else {
+				// If the file is not available after the maximum number of retries, show that it will be emailed.
+				setIsDownloading( false );
+				createNotice(
+					'success',
+					sprintf(
+						__(
+							'Your export will be emailed to %s',
```

Comment:
Need to review this wording too – more clearly explain what's happening. 

Something like `Your CSV export is being processed. It will be sent to <email> within 24 hours`.

Reply:
> I don't think we need to scare the users with the 24h waiting period (I assume realistically in major cases it will happen faster).

For sure, that is a placeholder for whatever a good value is. Sounds like we're broadly aligned @rogermattic it's just getting the details of the flow and copy right 👌🏻